### DISCUSSION TITLE
Running multiple pb instance for different pb_data directories

### DISCUSSION THREAD

**Discussion by rameshkumar70** [pocketbase/pocketbase#5255] [Q&A] [answered]
Hi, 
I wish to have a separate database i.e. pb_data folder, for each usage scenario of production, dev and testing. In the ticket https://github.com/pocketbase/pocketbase/discussions/738 there is a recommendation for running three separate pocketbase processes pointing to three different folders:

As an example:
pocketbase --dir pb_data  --http 'localhost:8091' &
pocketbase --dir pb_data_dev  --http 'localhost:8092' &
pocketbase --dir pb_data_test  --http 'localhost:8093' &

Suppose I have the following directory structure:
.
├── pb_data
├── pb_data_dev
├── pb_data_test
├── pb_migrations
└── pocketbase

How will pb_migrations be affected? Would pb_migrations be affected regardless of whichever pb_data* folder I alter?

Many thanks in advance.

**Comment by ganigeorgiev** [comment, marked as answer]
In that case the `pb_migrations` directory (and `pb_public` and `pb_hooks`) will be shared across all instances. If you want all applications to share the same collections configuration and custom code then this should be fine, otherwise you'll have to either:
- specify the `--hooksDir`, `--migrationsDir`, `--publicDir` flags (_use `--help` to explore all flags_)
- OR have multiple executables in separate directories for each app, e.g. app1, app2, app3 (_this is usually how you'll have to do it if you extend PocketBase with Go or when your apps uses different PocketBase versions_).

**Reply by rameshkumar70** [reply]
many thanks
